spring-cloud-gcp-data-spanner/src/main/java/org/springframework/cloud/gcp/data/spanner/core/SpannerTemplate.java

```diff
@@ -213,9 +216,11 @@ public long executePartitionedDmlStatement(Statement statement) {
 			SpannerPersistentEntity<T> persistentEntity) {
 		//an entity is eligible if all of the following true:
 		//1. entity has eager loaded properties
-		//2. there are no read options, as they can't be applied to a query
-		//3. key set does not have ranges, as they can't be used in a query
+		//2. entity has "Where" annotation
+		//3. there are no read options, as they can't be applied to a query
+		//4. key set does not have ranges, as they can't be used in a query
 		return persistentEntity.hasEagerlyLoadedProperties() &&
+				persistentEntity.findAnnotation(Where.class) == null &&
```

Comment:
create `getWhere()` method in `SpannerPersistentEntity`

Reply:
clear